.github/workflows/manifest-checks.yaml

```
@@ -0,0 +1,135 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # Fetch both the merge commit GitHub generates for the pull request
+          # and both its parents, i.e. the tip of the branch and the tip of main
+          fetch-depth: 2
+      - name: install `kustomize-build-dirs`
+        run: |
+          curl \
+            --location \
+            --silent \
+            https://github.com/utilitywarehouse/manifest-checkers/releases/download/v0.1.0/manifest-checkers_Linux_x86_64.tar.gz \
+          | tar \
+            --directory /usr/local/bin \
+            --extract \
+            --gzip \
+            --file - \
+            kustomize-build-dirs \
+      - name: Ensure Kustomize
+        run: >-
+          command -v kustomize ||
+          curl --silent --location https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv5.2.1/kustomize_v5.2.1_linux_amd64.tar.gz |
+          tar zx -C /usr/local/bin
+      - name: Run `kustomize build` across changed directories
+        run: |
+          set -o pipefail
+          mkdir built-manifests/
+          git ls-files -- '*/kustomization.yaml' | xargs kustomize-build-dirs --out-dir built-manifests/ --truncate-secrets 2>&1 \
+            | tee build-output
+      - name: Check for build errors
+        id: check-errors
+        run: |
+          # grab text between delimiters "---End warnings---"/"---start Warnings---"
+          build_errors="$(awk '$0=="---End warnings---"{d=0} d{print $0} $0=="---start Warnings---"{d=1}' build-output)"
+          if [ -n "$build_errors" ]
+          then
+            # multi-line output https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-of-a-multiline-string
+            echo "build-errs<<EOF" >> "$GITHUB_OUTPUT"
+            # print lines until the line that takes the total output is over 25000 characters
+            # this is to keep output size small while also not cutting things off mid-line (which can lead to confusing output)
+            awk -v max_length=25000 '{len+=length(); print} len >= max_length {exit(0)}' <<< "${build_errors}" >> "$GITHUB_OUTPUT"
+            echo -e "\n=============================================" >> "$GITHUB_OUTPUT"
+            if [[ ${#build_errors} -gt 25000 ]]
+            then
+              echo -e "\n(Errors output is truncated to 25000 characters)" >> "$GITHUB_OUTPUT"
+            fi
+            echo "EOF" >> "$GITHUB_OUTPUT"
+          fi
+      - name: Run `kustomize build` across files in main
+        run: |
+          set -o pipefail
+
+          # checkout the first parent of the merge commit (main)
+          # and build against the changes on the current branch
+          mkdir root-manifests/
+          pr_head="$(git rev-parse --verify HEAD)"
+          main_sha="$(git rev-parse --verify HEAD^1)"
+          git -c advice.detachedHead=false checkout --force "$main_sha"
```

I think we can simplify this 

I think we can simplify this now we're not building the list of changed files

```suggestion
          # main is the first parent of the merge commit created by GitHub
          main_sha="$(git rev-parse --verify HEAD^1)"
          git -c advice.detachedHead=false checkout --force "$main_sha"
```

---

But the directory root-manifests contains all of the built `main` manifests, used [here](https://github.com/utilitywarehouse/shared-kustomize-bases/blob/DENA-489/.github/workflows/manifest-checks.yaml#L78)- I think it is convenient and clean way to have two versions of manifests generated, in order to compare them.

---

> I think it is convenient and clean way to have two versions of manifests generated, in order to compare them.

we're not using `pr_head`, we can remove it, we don't need to make `root-manifests`, `build-kustomize-dirs` will create it if it doesn't exist. We will still have the manifests built for what's in `main`

---

[Applied](https://github.com/utilitywarehouse/shared-kustomize-bases/pull/30/commits/ed79caf0b248f988d819743788db25b0f1f71dce)